### ISSUE TITLE
Host ability to disable UI / put card into a pending state when card action is in progress

### ISSUE THREAD

**Issue by andrewleader** [microsoft/AdaptiveCards#1723]
# Problem

Some hosts want to display progress UI or disabled UI after the user clicks an action and the action is being performed.

# Asks

* Cortana

**Comment by Kartik1607** [comment]
Also, it would be nice if we could disable UI of actions, after an action has been performed. Something like a single click action.

There has been cases with out chat bot, that users clicks on submit button of adaptive card multiple times, causing the message to leak to next waterfall in bot framework, where the message throws up an error.

**Comment by andrewleader** [comment]
@Kartik1607 what if we just disable the entire card UI? There can be actions on the card outside of the action buttons, for example looks in text or clickable images.

**Comment by dclaux** [comment]
FYI Outlook Actionable Messages use this model:
![image](https://user-images.githubusercontent.com/1334689/43154930-c3b0b42e-8f2a-11e8-8975-2a86b3e949e6.png)

We disable the whole card, dim it and overlay a progress indicator (actual visuals are slightly different, this is just a mock from my spec.)

**Comment by Kartik1607** [comment]
@andrewleader If whole card is disabled, the actions on images would also be disabled. Is there any situation where one would like some particular actions to not be disabled, while would like to disable certain other actions. 

We have an adaptive card, where some of the images have action to open URL in other tab (app store, play store banners), which should be clickable any number of times.

Also, there are cards through which we receive user feedback. Since, adaptive cards currently give no feedback that a button has been pressed, we have lots of users who press the same button many times, and the event leaks to next dialog in the waterfall. So, these need to be clickable only once, and should not be re-enabled again.

**Comment by andrewleader** [comment]
@Kartik1607 you're asking for the ability to permanently disable an action after it has been clicked? 

As in, there's two buttons, "Yes" and "No", user clicks "Yes" and then "Yes" becomes permanently disabled? But then user can still click no?

"No" should probably also be disabled in that cass. That would likely have to be developer defined, since there's no way for the card, or the host, to know whether one button is related to the other button.

Ideally, the card author probably wants to replace the card and remove the buttons and say "Thanks for voting".

Is this the scenario you're trying to address?

That's a different scenario than what we're talking about here... What we're talking about is User clicks "Yes" which performs an async action, so the card should be put into a pending unclickable state until the action completes.

**Comment by Kartik1607** [comment]
> "No" should probably also be disabled in that cass. 

Yup

> Ideally, the card author probably wants to replace the card and remove the buttons and say "Thanks for voting"

That's a fantastic idea. I think a middle ground would be something like, when a user clicks a button we show a progress bar, which disables the whole card  (like in above image). When that task is completed, can re replace/remove some parts of the adaptive card (something like an svg is animated)

> That would likely have to be developer defined, since there's no way for the card, or the host, to know whether one button is related to the other button.

Agreed.

**Comment by Kartik1607** [comment]
@dclaux YUP

![Exactly](https://vignette.wikia.nocookie.net/cookieclicker/images/e/e8/Exactly.gif/revision/latest?cb=20140309024441)

**Comment by andrewleader** [comment]
Created a new issue to track author's ability to use a "refresh model" upon action click: #1737

**Comment by paulcam206** [comment]
Not currently under consideration, but will reactivate should we decide to pursue.
